### PR TITLE
Feat: Sanctum

### PR DESCRIPTION
**Pull request by Setanna**

### CODE REVIEW

**Comment by Setanna** [comment]
[API - Sanctum](https://trello.com/c/f6gTqrT5/4-api-sanctum)

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -103,4 +106,103 @@ public function getUsersByLocation(Location $location): \Illuminate\Http\Resourc
...
+    }
+
+    /**
+     * Check users token for ability
+     *
+     * @param Request $request
+     * @param $ability
+     * @return JsonResponse
+     */
+    public function checkAbility($ability) // TODO: RETURNS RECORD NOT FOUND
```
Comment (MathiasDuus) [inline app/Http/Controllers/UserController.php]: Oh no it is a wild TODO.

File: database/seeders/DatabaseSeeder.php [hunk 2]
```diff
@@ -16,6 +18,21 @@ class DatabaseSeeder extends Seeder
...
+            $Model->save();
+        }
+
+        Ability('change email');
+        Ability('password reset');
+        Ability('create');
+        Ability('read');
+        Ability('update');
+        Ability('destroy');
+        Ability('crud auth');
```
Comment (MathiasDuus) [inline database/seeders/DatabaseSeeder.php]: Better name

File: routes/api.php [hunk 3]
```diff
@@ -9,13 +9,67 @@
...
+    Route::put('/roles/{role}', [RoleController::class, 'update']);
+    /* Destroy */
+    Route::delete('/users/{user}', [UserController::class, 'destroy']);
+    Route::delete('/roles/{role}', [RoleController::class, 'destroy']);
+});
+
+/* Unprotected Routes */
+
+/* Read - Get */
+Route::group(['middleware' => ['api']], function () {
```
Comment (MathiasDuus) [inline routes/api.php]: Does it need the api middleware?

**Review by MathiasDuus — CHANGES_REQUESTED** [review]
While I don't approve of editing old migration files that have been run and manually changing the order of the files. I can see that it might have been necessary.
But in genneral don't add to existing migration files, add new ones and add the stuff in these. 
So that it is not required to run migrate:fresh.

File: routes/api.php (hunk 3 above)
Comment (Setanna) [reply routes/api.php]: Redundant route group has been removed

File: database/seeders/DatabaseSeeder.php (hunk 2 above)
Comment (Setanna) [reply database/seeders/DatabaseSeeder.php]: Parted the name "crud auth" into "create roles and users", "read roles and users", "update roles and users" and "destroy roles and users"

File: app/Http/Controllers/UserController.php (hunk 1 above)
Comment (Setanna) [reply app/Http/Controllers/UserController.php]: Removed outdated TODO

File: app/Http/Controllers/UserController.php [hunk 4]
```diff
@@ -103,4 +106,103 @@ public function getUsersByLocation(Location $location): \Illuminate\Http\Resourc
...
+            }
+            return response()->json([
+                'has_ability' => false
+            ]);
+        } else {
+            return response()->json([
+                'message' => 'Unauthenticated',
+            ], 403);
+        }
+    }
```
Comment (MathiasDuus) [inline app/Http/Controllers/UserController.php]: I think these out commented returns should be removed.

**Review by MathiasDuus — CHANGES_REQUESTED** [review]
Only a minor change, else it looks good :)

File: app/Http/Controllers/UserController.php (hunk 4 above)
Comment (Setanna) [reply app/Http/Controllers/UserController.php]: Unused code removed and replaced with minor documentation

**Review by MathiasDuus — APPROVED** [review]
LGTM 👍

**Pull request merged**